ros_buildfarm/config/doc_build_file.py

```diff
@@ -72,9 +74,16 @@ def __init__(self, name, data):
             self.doc_repositories = data['doc_repositories']
             assert isinstance(self.doc_repositories, list)
 
-        # doc_repositories can only be used with doc type make_target
+        # doc_repositories can only be used with make_target and external_site
+        # doc types
         is_make_target_type = self.documentation_type == DOC_TYPE_MAKE
-        assert not self.doc_repositories or is_make_target_type
+        is_external_site_type = self.documentation_type == DOC_TYPE_EXTERNAL
+        are_doc_repositories_allowed = (is_make_target_type or
+                                        is_external_site_type)
+        assert not self.doc_repositories or are_doc_repositories_allowed
+
+        # Single doc_repository allowed for external_site doc type
+        assert not is_external_site_type or len(self.doc_repositories) == 1
```

Comment:
Why is this limitation necessary? Wouldn't it be easy & convenient if multiple repositories could be listed as for doc independent one?

Reply:
It occurred to me to raise this point as well. It would be reasonable to expect that one job could clone N doc_repositories then build and run N containers, even if the N we need is one.

I didn't raise this question as the use of the jenkins git publisher means that all containers be required to transform the same source repository, but that isn't actually different from all the doc jobs being published to the same rsync destination.

So unless there's something else Dirk and I are missing I think lifting this limitation and iterating over all doc_repositories to build their containers is reasonable.

Reply:
@nuclearsandwich @dirk-thomas AFAIK, and correct me if I'm wrong, but jenkins git publisher will only allow us to handle one destination repo, that in the case of multiple repos would not be the (most common) case. I'd expect each to commit to the same gh-pages branch in different repositories.

(I'm addressing comments, will reply the rest today)

Reply:
I think you're right about the limitation, but the existing independent packages jobs all push to the same rsync target. A set of external sites could potentially all push to subdirectories of the same destination repository and that could be useful.

Reply:
Hmm, I had not considered that possibility. Alright, fair enough, will add that then.